pubspec.yaml

```diff
@@ -1,13 +1,13 @@
 name: http_multi_server
-version: 2.1.1-dev
+version: 2.2.0-dev
 
 description: >-
   A dart:io HttpServer wrapper that handles requests from multiple servers.
 author: Dart Team <misc@dartlang.org>
 homepage: https://github.com/dart-lang/http_multi_server
 
 environment:
-  sdk: '>=2.1.1 <3.0.0'
+  sdk: '>=2.8.0.dev.20.0 <3.0.0'
```

Comment:
```suggestion
  sdk: '>=2.8.0-dev.20.0 <3.0.0'
```

Reply:
Where did this version number come from? It's not out yet.

If we want to roll this out more incrementally we can add it to the signature now, but not forward it. The SDK constraint can be kept the same and we can publish. Then once there is a published SDK we can add the forwarding and bump the min SDK at that point.

Reply:
The purpose of using a future version number is to prevent this breaking users. The other reason is I don't know when my change will be rolled into a stable release, so I can't put a concrete version number.

I'm not quite sure how publication works. But I don't want this package to be published until my change is released.

I need the commit hash of this pr so that my change in SDK can build successfully. So I assume I need to land this first before I update DEPS file in SDK.

Can we have this landed but not published? Is it possible?

Reply:
> Can we have this landed but not published? Is it possible?

We wouldn't want it in that state for very long. We definitely shouldn't use an arbitrary SDK constraint to express that this can't go out until some point. If we want to express "it's not safe to publish this" then we would us `publish_to: none` with a comment explaining why. I don't think we should need to do that.

What do you think of my proposed roll out plan? It should be non breaking to add it to the signature without forwarding, and that's backwards and forwards compatible. We can roll that internally so that it's not coupled with the SDK roll. We can roll that to the SDK. We won't be blocked from publishing at any time if there is a need to do so. Once we have a known supported SDK version we can commit with a correct SDK constraint and continue to not be blocked from publishing.

Reply:
> We wouldn't want it in that state for very long. 

Definitely agree. It will be landed in next release if everything works smoothly. But I'm not sure whether it will be next few days or weeks.

> What do you think of my proposed roll out plan? It should be non breaking to add it to the signature without forwarding, and that's backwards and forwards compatible. We can roll that internally so that it's not coupled with the SDK roll. We can roll that to the SDK. We won't be blocked from publishing at any time if there is a need to do so. Once we have a known supported SDK version we can commit with a correct SDK constraint and continue to not be blocked from publishing.

Sgtm! But what do you mean by "roll that internally"? Is this change visible outside once we land it? Or Is there any intermediate stage between landing and publishing?

Reply:
> what do you mean by "roll that internally"?

I mean roll it in to google3. If we don't do an incremental migration then the SDK roll and the package roll would need to be in the same CL.